### PR TITLE
VZ-10280: Have VZ analyze process Rancher cluster management resources

### PR DESCRIPTION
**Pull request by mgianatagh**
Provide a summary of the change and which issue (i.e. ticket) is fixed.
If there are any dependencies, for example on a PR in another repository, list those.

### CODE REVIEW

File: tools/vz/pkg/analysis/internal/util/cluster/rancher.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+	var issueReporter = report.IssueReporter{
+		PendingIssues: make(map[string]report.Issue),
+	}
+
+	return analyzeRancherClusters(log, clusterRoot, &issueReporter)
+}
+
+// analyzeRancherClusters - analyze the status of Rancher clusters
+func analyzeRancherClusters(log *zap.SugaredLogger, clusterRoot string, issueReporter *report.IssueReporter) error {
+	clusterPath := files.FindFileInClusterRoot(clusterRoot, "default/cluster.json")
```
Comment (bobdonat) [inline tools/vz/pkg/analysis/internal/util/cluster/rancher.go:56]: Do we actually capture this file?

Comment (mgianatagh) [reply tools/vz/pkg/analysis/internal/util/cluster/rancher.go:56]: We do as of this PR

File: tools/vz/pkg/analysis/internal/util/cluster/rancher.go [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+	}
+
+	issueReporter.Contribute(log, clusterRoot)
+	return nil
+}
+
+// analyzeRancherCluster - analyze a single Rancher cluster and report any issues
+func analyzeRancherCluster(clusterRoot string, cluster rancherCluster, issueReporter *report.IssueReporter) error {
+
+	var messages []string
```
Comment (bobdonat) [inline tools/vz/pkg/analysis/internal/util/cluster/rancher.go:92]: Which cluster resource is this?  I have 5 cluster.xxx crds on my cluster.  Should the api group be checked to make it is the correct type of cluster resource

File: tools/vz/pkg/analysis/internal/util/cluster/rancher.go (hunk 1 above)
Comment (bobdonat) [reply tools/vz/pkg/analysis/internal/util/cluster/rancher.go:56]: Missed that.

Comment (bobdonat) [reply tools/vz/pkg/analysis/internal/util/cluster/rancher.go:56]: I noticed that we are capturing this resource in the default namespace when it is a cluster resource.  Same was true for the other CAPI cluster resources.  I was expecting to not see a namespace for the cluster resources.

File: tools/vz/pkg/analysis/internal/util/cluster/rancher.go (hunk 2 above)
Comment (mgianatagh) [reply tools/vz/pkg/analysis/internal/util/cluster/rancher.go:93]: The content of clusters.json is guaranteed to have "clusters.management.cattle.io", that is what vz capture will do.  Should I change the name of the file to something else?  Like cattleclusters.json?

**Review by bobdonat — APPROVED** [review]

**Pull request merged**